File: org.springframework.ide.eclipse.boot.dash/employee-service/src/main/java/com/bank/employee/service/EmployeeService.java
package com.bank.employee.service;

import com.bank.employee.entity.Employee;
import java.util.List;

public interface EmployeeService {
    Employee create(Employee employee);
    List<Employee> getAll();
    Employee getById(Long id);
    Employee login(String email, String password);
    Employee update(Long id, Employee employee);
    void delete(Long id);
}

File: banking-ui/src/app/service/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Employee {
  id?: number;
  name: string;
  email: string;
  password: string;
  role: string;
  salary: number;
  status?: string;
  joinedDate?: string;
}

@Injectable({ providedIn: 'root' })
export class EmployeeService {
  private baseUrl = 'http://localhost:8083/employees';

  constructor(private http: HttpClient) {}

  create(e: Employee): Observable<Employee> {
    return this.http.post<Employee>(this.baseUrl, e);
  }

  getAll(): Observable<Employee[]> {
    return this.http.get<Employee[]>(this.baseUrl);
  }

  delete(id: number): Observable<string> {
    return this.http.delete<string>(`${this.baseUrl}/${id}`);
  }
}
